internal/cli/watch_opts.go

```diff
@@ -55,6 +67,27 @@ func (opts *WatchOpts) Watch(f Watcher) error {
 	}
 }
 
+func (opts *WatchOpts) exponentialBackoff(err error) bool {
+	if opts.n <= 8 && checkForError(err, "CLUSTER_NOT_FOUND") {
+		backoff := math.Pow(base, float64(opts.n))
+		opts.n *= 2
+		sleepTime := time.Duration(backoff) * time.Second
```

Comment:
I think your logic seems a bit complicated to follow.

I had to follow the initial of 2 at line 47, the base at line 36 and the calculations at lines 72 & 74, the step of your loop at line 73, and finally the loop stop condition at line 71.

Perhaps we could try a logic without calculations how about something like `backoffTimes := []time.Duration(x * time.Seconds, y * time.Seconds, ...)` and a simple `n.i = 0` and a step by one. I think it would greatly improve code readability.

Reply:
Good point. I removed the part with calculations, and added `backoffTimes` to the `exponentialBackoff` method. I think it should be easier to understand now.